odoo/addons/base/models/ir_actions.py

```diff
@@ -457,6 +460,86 @@ def _get_readable_fields(self):
             "target", "url", "close",
         }
 
+
+class ServerActionHistoryWizard(models.TransientModel):
+    """ A wizard to compare and reset server action code. """
+    _name = 'server.action.history.wizard'
+    _description = "Server Action History Wizard"
+
+    @api.model
+    def _default_action_id(self):
+        return self.env['ir.actions.server'].browse(self._context.get('action_id', False))
+
+    action_id = fields.Many2one('ir.actions.server', default=_default_action_id)
+    code_diff = fields.Html(compute='_compute_code_diff', sanitize_tags=False)
+    revision = fields.Many2one("ir.actions.server.history",
+        domain="[('action_id', '=', action_id), ('active', '=', False)]",
+        default=lambda self: self.env["ir.actions.server.history"].search(
+            [("action_id", "=", self._context.get("action_id", False)), ("active", "=", False)],
+            limit=1,
+        ),
+        required=True,
+    )
+
+    @api.depends("revision")
+    def _compute_code_diff(self):
+        for wizard in self:
+            rev_code = wizard.revision.code
+            actual_code = wizard.action_id.code
+            has_diff = actual_code != rev_code
+            wizard.code_diff = get_diff(
+                    (actual_code or "", _("Actual Code")),
+                    (rev_code or "", _("Revision Code")),
+                    dark_color_scheme=request and request.cookies.get("color_scheme") == "dark",
+            ) if has_diff else False
+
+    def restore_revision(self):
+        self.ensure_one()
+        self.action_id.code = self.revision.code
+
+
+class IrActionsServerHistory(models.Model):
+    _name = 'ir.actions.server.history'
+    _description = 'Server Action History'
+    _order = 'code_date desc, id desc'
+    _log_access = False
+    _max_entries_per_action = 100
+
+    action_id = fields.Many2one('ir.actions.server', required=True, ondelete='cascade')
+    active = fields.Boolean(string='Current Revision', default=True)
```

Comment:
Is this field needed? A history that has only the last item that is active is not a history in my point of view.